### ISSUE TITLE
Add subtitle file failed

### ISSUE THREAD

**Issue by 20NE** [alexballas/go2tv#61]
There is option "-s" to add subtitles file in CLI mode, but doesn't work.
I tried: 

- rename subtitle filename to the same as video filename.
- add "-tc" option.
- on http url media or local video file.

All failed.
It seems DLNA doesn't support subtitle.

**Comment by alexballas** [comment]
Hello, I tested it and it works fine on my TV. What device are you using? (model name etc)
Maybe this is a limitation with your TV / Device since not all devices support DLNA subtitles.

**Comment by 20NE** [comment]
Xiaomi TV, I haven't tried on any other device yet

**Comment by alexballas** [comment]
Unfortunately this looks like a Xiaomi specific issue. Go2TV uses UPNP/DLNA to communicate and cast media to the devices. Unfortunately subtitles is not part of UPNP specification. There are some vendors however that introduced some proprietary extensions to the protocol to support subtitles like Samsung and Panasonic but this is not the case for every TV out there.

It would be good if you could test it with a different device. I personally tested it on Samsung and LG TVs and the subtitles work fine.

**Comment by nakhla** [comment]
sorry for commenting to a closed issue, but i think it is related.
I tried go2tv on bubbleupnp and it works for video and subtitle but for samsung j5200 subtitle not shown
i have tried gerbera and it works on same tv for video and subtitle, i am available if i can do any other tests.
thanks

**Comment by alexballas** [comment]
Hello @nakhla,
Unfortunately there is very little here to work with as I personally don't own a J5200. Do you have any other device you could test it with?
I'm also not aware how gerbera does it. You could potentially sniff the XML requests that gerbera sends to your device and see how the subtitle were implemented in the gerbera code.

**Comment by nakhla** [comment]
yes i tested with a tu7000 and go2tv is working for video and subtitle flawlessly, i make further test with gerbera on tu7000 and found that subtitle is not working.
so when gerbera work go2tv is not and when go2tv work gerbera is not working.
and i noticed go2tv is working in newly manufactured tvs and gerbera working in quiet old tvs

**Comment by alexballas** [comment]
FYI, I've been working on some improvements around transcoding (requires ffmpeg). With the upcoming version of Go2TV and assuming you have ffmpeg installed or the version from flathub, you should be able to bake the subtitles into the video. Stay tuned :)

**Comment by oyyh7** [comment]
我的字幕推送也无效，加载的字幕无法在电视上显示。
![Image](https://github.com/user-attachments/assets/fbb209cb-4fd7-4dda-a09a-a2646bbb214a)
![Image](https://github.com/user-attachments/assets/a791a557-3043-4e64-b13b-2610e93cbe13)

**Comment by alexballas** [comment]
> 我的字幕推送也无效，加载的字幕无法在电视上显示。
Translation: My subtitle push is also not working, the loaded subtitles cannot be displayed on the TV.

Hello @oyyh7,
With the latest version 1.8.0 if you check the transcode checkbox (转码) you can bake the subtitles into the video as an alternative

**Comment by oyyh7** [comment]
> > 我的字幕推送也无效，加载的字幕无法在电视上显示。
> > Translation: My subtitle push is also not working, the loaded subtitles cannot be displayed on the TV.
> 
> Hello [@oyyh7](https://github.com/oyyh7), With the latest version 1.8.0 if you check the transcode checkbox (转码) you can bake the subtitles into the video as an alternative

勾选了“转码”功能，也无法显示，甚至视频都无法在电视上播放了。

**Comment by nakhla** [comment]
> > 我的字幕推送也无效，加载的字幕无法在电视上显示。
> > Translation: My subtitle push is also not working, the loaded subtitles cannot be displayed on the TV.
> 
> Hello [@oyyh7](https://github.com/oyyh7), With the latest version 1.8.0 if you check the transcode checkbox (转码) you can bake the subtitles into the video as an alternative

i tried version 1.18 with transcode and it worked, as expected more cpu usage , but my concern now is i want to get change font size and type, now the subtitles are baked inside video, is there away to change font setting while baking??

**Comment by alexballas** [comment]
> > > 我的字幕推送也无效，加载的字幕无法在电视上显示。
> > > Translation: My subtitle push is also not working, the loaded subtitles cannot be displayed on the TV.
> > 
> > 
> > Hello [@oyyh7](https://github.com/oyyh7), With the latest version 1.8.0 if you check the transcode checkbox (转码) you can bake the subtitles into the video as an alternative
> 
> i tried version 1.18 with transcode and it worked, as expected more cpu usage , but my concern now is i want to get change font size and type, now the subtitles are baked inside video, is there away to change font setting while baking??

1.18 does not offer the ability to change the subtitle size in ffmpeg. I cound expose a config in the future to enabled this functionality but that would be only specific to ffmpeg and it could confuse people trying to get different subtitle sizes when casting without ffmpeg.
